File: verityx/src/main/java/com/example/verityx/service/SysPermissionService.java
package com.example.verityx.service;

import com.example.verityx.dto.SysPermissionDTO;
import com.example.verityx.entity.SysPermission;

import java.util.List;

/**
 * 系统权限服务接口
 */
public interface SysPermissionService {
    
    /**
     * 查询所有权限列表
     *
     * @return 权限列表
     */
    List<SysPermissionDTO> getAllPermissions();
    
    /**
     * 根据权限类型查询权限列表
     *
     * @param permissionType 权限类型
     * @return 权限列表
     */
    List<SysPermissionDTO> getPermissionsByType(String permissionType);
    
    /**
     * 根据ID查询权限详情
     *
     * @param id 权限ID
     * @return 权限详情
     */
    SysPermissionDTO getPermissionById(Long id);
    
    /**
     * 根据权限编码查询权限
     *
     * @param permissionCode 权限编码
     * @return 权限信息
     */
    SysPermission getPermissionByCode(String permissionCode);
    
    /**
     * 创建权限
     *
     * @param permission 权限信息
     * @param operator   操作人ID
     * @return 权限ID
     */
    Long createPermission(SysPermission permission, Long operator);
    
    /**
     * 更新权限
     *
     * @param permission 权限信息
     * @param operator   操作人ID
     * @return 是否成功
     */
    boolean updatePermission(SysPermission permission, Long operator);
    
    /**
     * 删除权限
     *
     * @param id       权限ID
     * @param operator 操作人ID
     * @return 是否成功
     */
    boolean deletePermission(Long id, Long operator);
    
    /**
     * 根据父ID查询权限列表
     *
     * @param parentId 父ID
     * @return 权限列表
     */
    List<SysPermissionDTO> getPermissionsByParentId(Long parentId);
    
    /**
     * 根据角色ID查询权限列表
     *
     * @param roleId 角色ID
     * @return 权限列表
     */
    List<SysPermissionDTO> getPermissionsByRoleId(Long roleId);
    
    /**
     * 根据用户ID查询权限列表
     *
     * @param userId 用户ID
     * @return 权限列表
     */
    List<SysPermissionDTO> getPermissionsByUserId(Long userId);
    
    /**
     * 查询菜单权限列表
     *
     * @return 菜单权限列表
     */
    List<SysPermissionDTO> getMenuPermissions();
    
    /**
     * 根据用户ID查询菜单权限列表
     *
     * @param userId 用户ID
     * @return 菜单权限列表
     */
    List<SysPermissionDTO> getMenuPermissionsByUserId(Long userId);
} 

File: front/src/api/sysPermissionService.js
import { http } from '@/utils/request'

export const SysPermissionApi = {
  // 获取所有权限列表
  getAllPermissions: () => {
    return http.get('/api/sys/permissions');
  },
  
  // 根据权限类型获取权限列表
  getPermissionsByType: (permissionType) => {
    return http.get(`/api/sys/permissions/type/${permissionType}`);
  },
  
  // 获取权限详情
  getPermissionById: (id) => {
    return http.get(`/api/sys/permissions/${id}`);
  },
  
  // 创建权限
  createPermission: (data) => {
    // 确保数据格式正确
    const permissionData = {
      permissionName: data.permissionName,
      permissionCode: data.permissionCode,
      permissionType: data.permissionType || 'menu',
      parentId: data.parentId ? Number(data.parentId) : 0,
      routePath: data.routePath || '',
      icon: data.icon || '',
      sort: data.sort ? Number(data.sort) : 0,
      isActive: data.isActive !== false,
      description: data.description || ''
    };
    
    return http.post('/api/sys/permissions', permissionData);
  },
  
  // 更新权限
  updatePermission: (data) => {
    // 确保数据格式正确
    const permissionData = {
      id: data.id ? Number(data.id) : null,
      permissionName: data.permissionName,
      permissionCode: data.permissionCode,
      permissionType: data.permissionType || 'menu',
      parentId: data.parentId ? Number(data.parentId) : 0,
      routePath: data.routePath || '',
      icon: data.icon || '',
      sort: data.sort ? Number(data.sort) : 0,
      isActive: data.isActive !== false,
      description: data.description || ''
    };
    
    return http.put('/api/sys/permissions', permissionData);
  },
  
  // 删除权限
  deletePermission: (id) => {
    return http.delete(`/api/sys/permissions/${id}`);
  },
  
  // 根据父ID获取权限列表
  getPermissionsByParentId: (parentId) => {
    return http.get(`/api/sys/permissions/parent/${parentId}`);
  },
  
  // 根据角色ID获取权限列表
  getPermissionsByRoleId: (roleId) => {
    return http.get(`/api/sys/permissions/role/${roleId}`);
  },
  
  // 根据用户ID获取权限列表
  getPermissionsByUserId: (userId) => {
    return http.get(`/api/sys/permissions/user/${userId}`);
  },
  
  // 获取菜单权限列表
  getMenuPermissions: () => {
    return http.get('/api/sys/permissions/menus');
  },
  
  // 根据用户ID获取菜单权限列表
  getMenuPermissionsByUserId: (userId) => {
    return http.get(`/api/sys/permissions/menus/user/${userId}`);
  }
};

export default SysPermissionApi; 